### ISSUE TITLE
Move flags excluding the build  per ghc version of plugins to plugins themselves

### ISSUE THREAD

**Issue by jneira** [haskell/haskell-language-server#2585]
* We have now the flags for including plugins in hls.cabal:

```
if flag(class) && (impl(ghc < 9.2.1) || flag(ignore-plugins-ghc-bounds))
    build-depends: hls-class-plugin ^>=1.0.0.1
    cpp-options: -Dclass
```

* that works for hls itself but you can still do `cabal build hls-class-plugin` using `ghc-9.2.1` and even worse, `cabal build all -w ghc-9.2.1` builds all packages including the plugins not buildable for that ghc version
* maybe we should move the `impl(ghc < 9.2.1) || flag(ignore-plugins-ghc-bounds` to each plugin .cabal and remove them for hls
  * it would make simpler the hls.cabal and put the flag close where thay are being effectively used

**Comment by michaelpj** [comment]
> maybe we should move the impl(ghc < 9.2.1) || flag(ignore-plugins-ghc-bounds to each plugin .cabal and remove them for hls 

I feel like I asked you about this at the time? I vaguely recall a couple of issues:
- Maybe it's not easy to set `buildable: false` in a conditional?
- Maybe it's helpful to still be able to build them, so you can check if they've started working?

**Comment by jneira** [comment]
Yeah, it has its downsides, and we talked about but i did not consider the `all` case.
One of the downsides is we will makeplugin developers to have to remember use `--ignore-ghc-bounds` when updating the plugin for the new ghc.
No problem in combine `buildable: false` with conditionals though.

So it is not clear, i think the right one would be `--enable-benchmarks` will work and maybe the workaround would be build the unique benchmark we care about right now, ghcide one and ya.
Closing then, thanks for your thoughts
